cmd/agent/app/check.go

```diff
@@ -127,19 +129,47 @@ var checkCmd = &cobra.Command{
 			fmt.Println("Multiple check instances found, running each of them")
 		}
 
+		var instancesData []interface{}
+
 		for _, c := range cs {
 			s := runCheck(c, agg)
 
 			// Sleep for a while to allow the aggregator to finish ingesting all the metrics/events/sc
 			time.Sleep(time.Duration(checkDelay) * time.Millisecond)
 
-			printMetrics(agg)
+			if formatJSON {
+				aggregatorData := getMetricsData(agg)
+				var collectorData map[string]interface{}
+
+				collectorJSON, _ := status.GetCheckStatusJSON(c, s)
+				json.Unmarshal(collectorJSON, &collectorData)
 
-			checkStatus, _ := status.GetCheckStatus(c, s)
-			fmt.Println(string(checkStatus))
+				checkRuns := collectorData["runnerStats"].(map[string]interface{})["Checks"].(map[string]interface{})[checkName].(map[string]interface{})
+
+				// There is only one checkID per run so we'll just access that
+				var runnerData map[string]interface{}
+				for _, checkIDData := range checkRuns {
+					runnerData = checkIDData.(map[string]interface{})
+					break
+				}
+
+				instanceData := map[string]interface{}{
+					"aggregator": aggregatorData,
+					"runner":     runnerData,
+				}
+				instancesData = append(instancesData, instanceData)
+			} else {
+				printMetrics(agg)
+				checkStatus, _ := status.GetCheckStatus(c, s)
```

Comment:
That's sad that even in the original code they were some errors silented on this.